File: backend/internal/database/model/warehouse.go
package model

import "time"

// Warehouse 仓库
type Warehouse struct {
	Id          int64     `json:"id" form:"id" db:"id"`
	Name        string    `json:"name" form:"name" db:"name"`
	Description *string   `json:"description" form:"description" db:"description"`
	CreateAt    time.Time `json:"create_at" form:"create_at" db:"create_at"`
}

File: frontend/src/types/warehouse.ts
/**
 * Mirrors backend model `internal/database/model/warehouse.go` and the
 * `warehouses` table migration. Keep in sync if the backend schema changes.
 *
 * Note: the backend model tags the timestamp as `json:"create_at"` (NOT
 * `created_at`) — the field is named `CreateAt` in Go. Keep this quirk in
 * sync here.
 *
 * `description` is `*string` (nullable) — null when unset.
 */
export interface Warehouse {
  id: number
  name: string
  description: string | null
  /** Backend serializes this as `create_at` (model.CreateAt). */
  create_at: string
}

/**
 * Payload for `POST /api/warehouses/register`. The backend binds
 * {name, description}; `name` is required, `description` is optional.
 */
export interface WarehouseInput {
  name: string
  description: string | null
}
